src/amici_hdf5.cpp

```diff
@@ -144,9 +151,10 @@ UserData *AMI_HDF5_readSimulationUserDataFromFileObject(hid_t fileId,
     if (AMI_HDF5_attributeExists(fileId, datasetPath, "pbar")) {
         status += AMI_HDF5_getDoubleArrayAttribute(fileId, datasetPath, "pbar",
                                                    &buffer, &length0);
-        if (length0 != 0 && length0 != (unsigned)udata->nplist)
+        if (length0 != 0 && length0 == (unsigned)udata->nplist())
             goto freturn;
-        udata->setPbar(buffer);
+        if(length0 == (unsigned)udata->nplist())
```

Comment:
We will never get there unless length0 == 0 and nplist == 0. Is this intended?

Reply:
I don't think so